src/server/shell/default_configuration.cpp

```diff
@@ -67,23 +65,27 @@ auto mir::DefaultServerConfiguration::the_window_manager_builder() -> shell::Win
 auto mir::DefaultServerConfiguration::the_decoration_manager() -> std::shared_ptr<msd::Manager>
 {
     return decoration_manager(
-        [this]()->std::shared_ptr<msd::Manager>
+        [this]
         {
-            return std::make_shared<msd::BasicManager>(
-                *the_display_configuration_observer_registrar(),
-                [buffer_allocator = the_buffer_allocator(),
-                 executor = the_main_loop(),
-                 cursor_images = the_cursor_images()](
-                    std::shared_ptr<shell::Shell> const& shell,
-                    std::shared_ptr<scene::Surface> const& surface) -> std::unique_ptr<msd::Decoration>
+            if (!decoration_manager_init)
```

Comment:
There doesn't need to be an `if` here - just a working default implementation. But you'll probably be reworking the logic following the "wrap_..." model.